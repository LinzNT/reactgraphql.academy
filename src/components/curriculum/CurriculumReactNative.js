import React from 'react'
import { H1Ref } from '../text'
import Link from '../navigation/Link'
import Section, { List, curriedToggleNavigateTo } from './CurriculumSection'
import { Col, Row } from '../layout/Grid'
import ReactNativeFoundationSession from './sessions/native/ReactNativeFoundationSession'
import ReactNativeNavigationSession from './sessions/native/ReactNativeNavigationSession'
import ReactNativeAnimationsSession from './sessions/native/ReactNativeAnimationsSession'
import { REACT_NATIVE } from '../../config/data'
import selectCurriculumLayout from './selectCurriculumLayout'

const CurriculumReactNative = ({
  showTitle = true,
  enableToggle,
  isOpen,
  toggleNavigateTo = `/curriculum?tab=${REACT_NATIVE}`,
  layout,
}) => {
  const toggleNavigateToSection = curriedToggleNavigateTo(toggleNavigateTo)
  const commonProps = {
    enableToggle,
    toggleNavigateTo: toggleNavigateToSection,
    type: REACT_NATIVE,
    isOpen,
  }
  const firstHalf = (
    <Section
      {...commonProps}
      title="React Native Day 1"
      name="day1"
      subTitle="Foundation, Navigation, and Animations"
    >
      <ReactNativeFoundationSession title="Foundation" />
      <ReactNativeNavigationSession title="Navigation" />
      <ReactNativeAnimationsSession title="Animations" />
    </Section>
  )

  return (
    <React.Fragment>
      {showTitle ? (
        <Row>
          <Col lg={10} lgOffset={1}>
            <H1Ref>
              React Native Curriculum
              <Link to="#curriculum" name="curriculum">
                #
              </Link>
            </H1Ref>
          </Col>
        </Row>
      ) : (
        ''
      )}
      {selectCurriculumLayout({ firstHalf, layout })}
    </React.Fragment>
  )
}

export default CurriculumReactNative
