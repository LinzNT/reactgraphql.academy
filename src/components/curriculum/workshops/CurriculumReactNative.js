import React from 'react'
import { H2Ref } from '../../text'
import Link from '../../navigation/Link'
import { Li } from '../../layout/Ul'
import Section, { curriedToggleNavigateTo } from '../CurriculumSection'
import ReactNativeFoundationSession from '../sessions/native/ReactNativeFoundationSession'
import ReactNativeNavigationSession from '../sessions/native/ReactNativeNavigationSession'
import ReactNativeAnimationsSession from '../sessions/native/ReactNativeAnimationsSession'
import { REACT_NATIVE } from '../../../config/data'
import selectCurriculumLayout from '../selectCurriculumLayout'

const CurriculumReactNative = ({
  showTitle = true,
  enableToggle,
  isOpen,
  toggleNavigateTo,
  showLinkToCurriculum = false,
  layout,
  trainings,
}) => {
  const toggleNavigateToSection = curriedToggleNavigateTo(toggleNavigateTo)
  const type = REACT_NATIVE
  const commonProps = {
    enableToggle,
    toggleNavigateTo: toggleNavigateToSection,
    type,
    isOpen,
  }
  const firstHalf = (
    <React.Fragment>
      <Section
        {...commonProps}
        title="React Native Day 1"
        name="day1"
        subTitle="Foundation, Navigation, and Animations"
      >
        <ReactNativeFoundationSession title="Foundation" />
        <ReactNativeNavigationSession title="Navigation" />
        <ReactNativeAnimationsSession title="Animations" />
      </Section>
    </React.Fragment>
  )

  const secondHalf = null

  const title = showTitle ? (
    <H2Ref>
      React Native Curriculum
      <Link to="#curriculum" name="curriculum">
        #
      </Link>
    </H2Ref>
  ) : null

  return selectCurriculumLayout({
    firstHalf,
    secondHalf,
    layout,
    type,
    title,
    trainings,
    curriculumTo: showLinkToCurriculum ? toggleNavigateTo : undefined,
  })
}

export const TargetAudienceList = () => (
  <React.Fragment>
    <Li>
      You have at least a few months of experience using React on the web. If
      you don't know React, we recommend you first to attend our{' '}
      <Link to="/react/training/bootcamp">React Bootcamp</Link>
    </Li>
    <Li>
      Taking a step forward to become a React Native Specialist able to make
      critical decisions about the architecture of a React Native application.
    </Li>
    <Li>
      Not satisfied with the pace of online learning and it's lack of 1-on-1
      mentoring?
    </Li>
  </React.Fragment>
)

export default CurriculumReactNative
