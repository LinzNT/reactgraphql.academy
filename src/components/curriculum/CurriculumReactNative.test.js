import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CurriculumReactNative from './workshops/CurriculumReactNative'

describe('<CurriculumReactNative />', () => {
  it('should render component', () => {
    const wrapper = shallow(<CurriculumReactNative />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
