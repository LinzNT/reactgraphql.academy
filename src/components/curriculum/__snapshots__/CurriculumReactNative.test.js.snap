// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurriculumReactNative /> should render component 1`] = `
<Fragment>
  <Row>
    <Styled(Col)
      lg={10}
      lgOffset={1}
    >
      <Styled(styled.h2)>
        React Native Curriculum
        <Link
          name="curriculum"
          to="#curriculum"
        >
          #
        </Link>
      </Styled(styled.h2)>
    </Styled(Col)>
  </Row>
  <Row>
    <Styled(Col)
      lg={5}
      lgOffset={1}
      md={6}
    >
      <withRouter(CurriculumSection)
        name="day1"
        subTitle="Foundation, Navigation, and Animations"
        title="React Native Day 1"
        toggleNavigateTo={[Function]}
        type="React Native"
      >
        <ReactNativeFoundationSession
          title="Foundation"
        />
        <ReactNativeNavigationSession
          title="Navigation"
        />
        <ReactNativeAnimationsSession
          title="Animations"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day2"
        subTitle="Gestures, Offline, Assets Management & Push Notifications"
        title="React Native Day 2"
        toggleNavigateTo={[Function]}
        type="React Native"
      >
        <ReactNativeGesturesSession
          title="Gestures"
        />
        <ReactNativeOfflineAndAssetCreationSession
          title="Handling Offline & Assets Management"
        />
        <ReactNativePushNotificationSession
          title="Push Notifications"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day3"
        subTitle="Testing, Native Modules & Release to Production"
        title="React Native Day 3"
        toggleNavigateTo={[Function]}
        type="React Native"
      >
        <ReactNativeTestingSession
          title="Testing in React Native"
        />
        <ReactNativeNativeModulesSession
          title="Native Modules"
        />
        <ReactNativeProductionSession
          title="Release to Production"
        />
      </withRouter(CurriculumSection)>
      <styled.div>
        <LinkButton
          secondary={true}
          to="/curriculum?tab=React Native"
          trackUserBehaviour={[Function]}
        >
          Full curriculum&gt;&gt;
        </LinkButton>
      </styled.div>
    </Styled(Col)>
    <Styled(Col)
      lg={5}
      lgOffset={1}
      md={6}
    >
      <UpcomingTrainingSection
        curriculum={true}
        type="React Native"
      />
    </Styled(Col)>
  </Row>
</Fragment>
`;
